TeamCode/src/main/java/org/firstinspires/ftc/teamcode/DriveTrain.java

```diff
@@ -0,0 +1,246 @@
+    LinearOpMode opMode;
+
+    //Hi, I'm adding this comment so I'll have made a change to my branch (TeleopDrivetranMechanum)
+
+    double wheelOnePower = 0.0; // 0.55 // The standard power for the wheels, will probably be changed later
+    double wheelTwoPower = 0.0; // 0.7
+    double wheelThreePower = 0.0; //0.35
+    double wheelFourPower = 0.0; // 0.2
+    double wheelSpeedChanger = 1.1;
+    double inchToTick = (360/6); // The number of encoder ticks per inch for our wheels
+    double sideInchToTick = (360/6); // The number of encoder ticks for one inch while travelling sideways, change later
+    double startPosition = 0;
+    double endPosition = 0;
+    double redLine = 0;  // One of the color sensor readings for the red line, definitely change later
+    double blueLine = 0; // Same as above but for the blue line
+    int counter = 0;
+    Orientation angles;
+    //BNO055IMU.Parameters parameters;
+
+    public DriveTrain(Telemetry telemetry, HardwareInnov8Robot robot, LinearOpMode opMode) {
+    /*
+        parameters = new BNO055IMU.Parameters();
+        parameters.angleUnit = BNO055IMU.AngleUnit.DEGREES;
+        parameters.accelUnit = BNO055IMU.AccelUnit.METERS_PERSEC_PERSEC;
+        parameters.calibrationDataFile = "BNO055IMUCalibration.json"; // see the calibration sample opmode
+        parameters.loggingEnabled = true;
+        parameters.loggingTag = "IMU";
+        parameters.accelerationIntegrationAlgorithm = new JustLoggingAccelerationIntegrator();
+    */
+// Retrieve and initialize the IMU. We expect the IMU to be attached to an I2C port
+// on a Core Device Interface Module, configured to be a sensor of type "AdaFruit IMU",
+// and named "imu".
+        this.opMode = opMode;
+        this.robot = robot;
+        this.telemetry = telemetry;
+       // this.robot.imu.initialize(parameters);
+       // angles = this.robot.imu.getAngularOrientation(AxesReference.INTRINSIC, AxesOrder.ZYX, AngleUnit.DEGREES);
+        this.telemetry.addData("first", angles.firstAngle);
+        this.telemetry.addData("second", angles.secondAngle);
+        this.telemetry.addData("third", angles.thirdAngle);
+        this.telemetry.addData("counter", counter);
+        this.telemetry.addData(DRIVE_TRAIN_CAPTION, "Drive train initialized");
+        this.telemetry.update();
+    }
+
+    public void goForward(double inches) {
+        Log.d(DRIVE_TRAIN_CAPTION, "Robot is moving forward");
+        this.telemetry.addData(DRIVE_TRAIN_CAPTION, "Robot is moving forward");
+        this.telemetry.addData("wheel power", this.robot.motorOne.getPower());
+        this.telemetry.update();
+        startPosition = this.robot.motorOne.getCurrentPosition();
+        endPosition = startPosition + (inches * inchToTick); // How far you need to travel
+        while (this.robot.motorOne.getCurrentPosition() < endPosition && this.opMode.opModeIsActive()) {
+            this.telemetry.addData("StartPosition", startPosition);
```

Comment:
We seem to send stuff both to telemetry and log.d it would be nice if we just had a function that did both so we can save lines of code